mkosi/machine.py

```diff
@@ -64,16 +68,36 @@ def __init__(self, args: Sequence[str] = [], debug: bool = False) -> None:
         tmp.autologin = True
         tmp.ephemeral = True
         if tmp.verb == Verb.qemu:
-            tmp.bootable = True
-            tmp.qemu_headless = True
             tmp.hostonly_initrd = True
             tmp.netdev = True
             tmp.ssh = True
         elif tmp.verb not in (Verb.shell, Verb.boot):
             die("No valid verb was entered.")
 
+        self.parsed_copy = copy.deepcopy(tmp)
+
+        # We want to make sure images which are not bootable are still built.
+        # Therefore, we overpass the actual verb passed by the user for now.
+        # Otherwise, load_args() raises an exception before the building is done.
+        # Then, when we actually boot the image, the user's verb is brought back.
+        v = tmp.verb
+        tmp.verb = Verb.build
+        if not tmp.with_unified_kernel_images and "uefi" in tmp.boot_protocols:
+            if tmp.distribution in (Distribution.debian, Distribution.ubuntu, Distribution.mageia, Distribution.opensuse):
+                tmp.with_unified_kernel_images = True
         self.args = load_args(tmp)
 
+        # Now we can use the verb passed by the constructor/environment variable.
+        # Exceptions about combinations of options not being bootable won't be raised.
+        self.args.verb = v
+
+        # Some checks don't take the Verb into account, but rather the attributes below.
+        if v == Verb.qemu:
+            self.parsed_copy.bootable = True
+            self.parsed_copy.qemu_headless = True
+            tmp.bootable = True
+            tmp.qemu_headless = True
```

Comment:
Just temporarily omitting the verb is not enough to built the image without raising exceptions in load_args() sometimes.